lib/mix/tasks/benchmark.ex

```diff
@@ -0,0 +1,199 @@
+defmodule Mix.Tasks.Benchmark.Helper do
+  @moduledoc false
+  defmacro compile_file(file) do
+    case compare_versions(System.version(), "1.7.0") do
+      :gt ->
+        quote do
+          Code.compile_file(unquote(file))
+        end
+
+      _ ->
+        quote do
+          Code.load_file(unquote(file))
+        end
+    end
+  end
+
+  defp compare_versions(v1, v2) do
+    Version.compare(Version.parse!(v1), Version.parse!(v2))
+  end
+end
+
+defmodule Mix.Tasks.Benchmark do
+  use Mix.Task
+  require Logger
+  alias Mix.Tasks.Benchmark.Helper
+  require Mix.Tasks.Benchmark.Helper
+
+  @moduledoc """
+  Runs project benchmarks
+  """
+  @dialyzer [no_match: [find_benchmark_files: 2]]
+
+  @switches []
+
+  @shortdoc "Runs project benchmarks"
+  @recursive true
+  @preferred_cli_env :test
```

Comment:
`:test` ? I wouldn't put any `cli_env` here

Reply:
Why not though?
There's a reason why environments exist, and one of such are dependencies. 
I believe you know about `:only` keyword for dependencies. 

For example, let's say one clones a project to run benchmarks locally on a machine. So, why do they need to download dependencies for `:prod` environment?

Reply:
I know about it, which is why we recommend [`only: :dev`](https://github.com/bencheeorg/benchee#installation) and hence I'd set that or nothing as preferred_cli_env.